common/src/main/java/org/opensearch/ml/common/agent/MLToolSpec.java

```diff
@@ -56,6 +70,9 @@ public MLToolSpec(StreamInput input) throws IOException {
             parameters = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
         }
         includeOutputInAgentResponse = input.readBoolean();
+        if (input.getVersion().onOrAfter(MINIMAL_SUPPORTED_VERSION_FOR_TOOL_CONFIG) && input.readBoolean()) {
+            configMap = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
```

Comment:
why `readOptionalString` ? so a key might not have a value?

Reply:
Yes, this allows you to override a value with a non-value.